ssize has not been declared in std

```c++
quickcpplib/include/quickcpplib/gsl-lite/include/gsl/gsl-lite.hpp:984:12: error: ‘ssize’ has not been declared in ‘std’
  984 | using std::ssize;
      |            ^~~~~
```
Cause:

```c++
#define gsl_HAVE_STD_SSIZE              ( gsl_COMPILER_GNUC_VERSION >= 1000 )
```
However `ssize()` is a C++ 20 only feature, not a GCC 10 feature. So this is the wrong check to do.

--- a/include/gsl/gsl-lite.hpp
+++ b/include/gsl/gsl-lite.hpp
@@ -394,7 +394,11 @@
 
 #define gsl_HAVE_CONTAINER_DATA_METHOD  gsl_CPP11_140_CPP0X_90
 #define gsl_HAVE_STD_DATA               gsl_CPP17_000
-#define gsl_HAVE_STD_SSIZE              ( gsl_COMPILER_GNUC_VERSION >= 1000 )
+#ifdef __cpp_lib_ssize
+# define gsl_HAVE_STD_SSIZE             1
+#else
+# define gsl_HAVE_STD_SSIZE             ( gsl_COMPILER_GNUC_VERSION >= 1000 && __cplusplus > 201703L )
+#endif
 
 #define gsl_HAVE_SIZED_TYPES            gsl_CPP11_140
 

